### PR TITLE
Feature/linkta 91 back end formats prompt and communicates with gemini

### PR DESCRIPTION
**Pull request by dcpavel**
## Description

This changes the /gen-ai/query route to convert the prompt into a form that will return a response from the LLM that can be used to generate a tree.

There are a number of prompt examples to use, which may need to be modified. This has been done so that testing can be implemented in order to expand types and test different prompts to determine the best results.

## Type of change

- [ ]  🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨  New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Has This Been Tested?

Manual testing.

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

## Additional Information:

There is work here that is meant to be built upon as needed in the future. A test of the current configuration gave this response:
"```JSON\n{\n  \"Architecture\": {\n    \"History\": [\n      \"Ancient Egypt\",\n      \"Ancient Greece\",\n      \"Ancient Rome\",\n      \"Byzantium\",\n      \"Renaissance\",\n      \"Baroque\",\n      \"Neoclassicism\",\n      \"Modernism\",\n      \"Postmodernism\"\n    ],\n    \"Building Types\": [\n      \"Residential\",\n      \"Commercial\",\n      \"Institutional\",\n      \"Civic\",\n      \"Industrial\",\n      \"Religious\",\n      \"Military\"\n    ],\n    \"Design Elements\": [\n      \"Form\",\n      \"Function\",\n      \"Structure\",\n      \"Materials\",\n      \"Light\",\n      \"Color\",\n      \"Texture\"\n    ],\n    \"Sustainable Design\": [\n      \"Green Building\",\n      \"Environmental Impact\",\n      \"Renewable Energy\",\n      \"Water Conservation\",\n      \"Waste Reduction\"\n    ],\n    \"Architectural Styles\": [\n      \"Classical\",\n      \"Gothic\",\n      \"Renaissance\",\n      \"Baroque\",\n      \"Neoclassicism\",\n      \"Modernism\",\n      \"Postmodernism\"\n    ]\n  }\n}\n```"

We are still not transforming this response into a tree, that is a separate issue.

### CODE REVIEW

**Comment by dcpavel** [comment]
Added the edge case I mentioned in standups.

File: server/utils/typeChecker.ts
Comment (kinskiwu) [inline server/utils/typeChecker.ts:1]: - Line 13: did you mean to return false when the lengths are not equal? 
- Consider renaming `isType` to a more descriptive name (sth like isValidAIInputType?) to more accurately reflect its role in validating complex input types for AI operations. 
- It appears that `isType` only checks the types for one layer of properties, do we also need the capability to check types for nested objects? I wonder if using a validation library like zod or joi may make type checking easier.
- Have we handled the edge case when prompt input is null? (in JS type of null is also an object)

File: server/controllers/genAiController.ts [hunk 1]
```diff
@@ -1,31 +1,48 @@
-import createAI from '@/server/utils/AIFactory';
+import createAI from '@server/utils/AIFactory';
 import { createError } from '@/server/middleware/errorHandling';
+import TreePrompts from '@/server/models/TreePromptsModel';
+import { isType } from '@server/utils/typeChecker';
+import { AIProvider } from '@server/types/index';
```
Comment (kinskiwu) [inline server/controllers/genAiController.ts:5]: Maybe `AIProvider` can be grouped with the import from line 8?

File: server/controllers/genAiController.ts
Comment (kinskiwu) [inline server/controllers/genAiController.ts:1]: Consider adding null checks and validate inputs for request payloads.

**Review by kinskiwu — APPROVED** [review]
Small changes. The rest looks good to me😊

File: server/controllers/genAiController.ts
Comment (dcpavel) [reply server/controllers/genAiController.ts:1]: Good idea.

File: server/utils/typeChecker.ts
Comment (LucieSeidler) [reply server/utils/typeChecker.ts:1]: second this question

**Review by LucieSeidler — CHANGES_REQUESTED** [review]

**Review by LucieSeidler — APPROVED** [review]

File: server/controllers/genAiController.ts (hunk 1 above)
Comment (kinskiwu) [reply server/controllers/genAiController.ts:5]: AIProvider is not a type. Disregard this comment.

**Pull request merged**